### ISSUE TITLE
Unaligned sequence support

### ISSUE THREAD

**Issue by SimonBartels** [MachineLearningLifeScience/poli#16]
The `AbstractBlackbox` currently asserts that the input is a `numpy` array with shape length 2. That's problematic for unaligned objectives. How about we instantiate black-boxes not with the sequence length but the problem information?

**Comment by miguelgondu** [comment]
Agreed. I find the checks we're performing on the `__call__` method highly problematic: they don't account for batched inputs, they don't allow for shape lengths larger than 2 (as you say), etc.

**Comment by miguelgondu** [comment]
We addressed with the refactoring you suggested, right? Let me know and I'll close!

We still check that `len(x.shape) == 2` (i.e. `[b, max_seq_length]`), though.
